pydantic/_internal/_typing_extra.py

```diff
@@ -265,24 +265,27 @@ def eval_type_backport(
                 value, globalns, localns
             )
     except TypeError as e:
-        if not (isinstance(value, typing.ForwardRef) and is_backport_fixable_error(e)):
-            raise
-        try:
-            from eval_type_backport import eval_type_backport
-        except ImportError:
-            raise TypeError(
-                f'You have a type annotation {value.__forward_arg__!r} '
-                f'which makes use of newer typing features than are supported in your version of Python. '
-                f'To handle this error, you should either remove the use of new syntax '
-                f'or install the `eval_type_backport` package.'
-            ) from e
-
-        return eval_type_backport(value, globalns, localns, try_default=False)
-
-
-def is_backport_fixable_error(e: TypeError) -> bool:
-    msg = str(e)
-    return msg.startswith('unsupported operand type(s) for |: ') or "' object is not subscriptable" in msg
+        if isinstance(value, typing.ForwardRef):
+            try:
+                from eval_type_backport import eval_type_backport
+            except ImportError:
+                raise TypeError(
+                    f'Unable to evaluate type annotation {value.__forward_arg__!r}. '
+                    'If you are making use of the new typing syntax (unions using `|` or builtins subscripting), '
+                    'you should either replace the use of new syntax with the existing `typing` constructs '
+                    'or install the `eval_type_backport` package. Otherwise, it might be '
+                    "that the type being used isn't subscriptable."
```

Comment:
Let's incorporate some of this old wording about Python versions:

```
raise TypeError(
                f'which makes use of newer typing features than are supported in your version of Python. '
            )
```

Otherwise, this looks good to me. Let's also fix the failing tests bc of the error wording mismatch 👍

Reply:
Should we even inject the Python version being used? With sys.version_info or something?

Reply:
Hum I think just mentioning the Python versions introducing the new features is fine?